pkgs/by-name/x2/x2t/package.nix

```diff
@@ -0,0 +1,344 @@
+    for i in ${openssl.out}/lib/*; do
+      ln -s $i Common/3dParty/openssl/build/linux_64/lib/$(basename $i)
+    done
+
+    echo "== v8 =="
+    mkdir -p Common/3dParty/v8_89/v8/out.gn/linux_64
+    ln -s ${v8}/lib Common/3dParty/v8_89/v8/out.gn/linux_64/obj
+    for i in ${v8.src}/*; do
+      ln -s $i Common/3dParty/v8_89/v8/$(basename $i)
+    done
+
+    echo "== doctrenderer =="
+    # requires graphics, openssl (with MD2), v8
+    cd DesktopEditor/doctrenderer
+    qmake "CONFIG+=debug" -o Makefile doctrenderer.pro
+    make -j$NIX_BUILD_CORES
+    cd ../..
+
+    echo "== HtmlRenderer =="
+    # requires UnicodeConverter, kernel, graphics
+    cd HtmlRenderer
+    qmake "CONFIG+=debug" -o Makefile htmlrenderer.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== HtmlFile2 =="
+    # depends on kernel, kernel_network, graphics and gumbo-parser
+    cd HtmlFile2
+    qmake "CONFIG+=debug" -o Makefile HtmlFile2.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== EpubFile =="
+    # requires kernel, graphics and HtmlFile2
+    cd EpubFile
+    qmake "CONFIG+=debug" -o Makefile CEpubFile.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== PdfFile =="
+    cd PdfFile
+    qmake "CONFIG+=debug" -o Makefile PdfFile.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== DjVuFile =="
+    # requires UnicodeConverter, kernel, graphics and PdfFile
+    cd DjVuFile
+    qmake "CONFIG+=debug" -o Makefile DjVuFile.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== XpsFile =="
+    # requires UnicodeConverter, graphics, kernel, PdfFile
+    cd XpsFile
+    qmake "CONFIG+=debug" -o Makefile XpsFile.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== Fb2File =="
+    cd Fb2File
+    qmake "CONFIG+=debug" -o Makefile Fb2File.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+
+    echo "== DocxRenderer =="
+    cd DocxRenderer
+    qmake "CONFIG+=debug" -o Makefile DocxRenderer.pro
+    make -j$NIX_BUILD_CORES
+    cd ..
+    
+    echo "== X2tConverter =="
+    cd X2tConverter/build/Qt
+    qmake "CONFIG+=debug" -o Makefile X2tConverter.pro
+    make -j$NIX_BUILD_CORES
+    cd ../../..
+    find .
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib
+    cp build/lib/linux_64/debug/*.so $out/lib
+
+    mkdir -p $out/bin
+    cp build/bin/linux_64/debug/x2t $out/bin
+
+    cat >$out/bin/DoctRenderer.config <<EOF
+      <Settings>
+        <file>${onlyoffice-documentserver}/var/www/onlyoffice/documentserver/sdkjs/common/Native/native.js</file>
+        <file>${onlyoffice-documentserver}/var/www/onlyoffice/documentserver/sdkjs/common/Native/jquery_native.js</file>
+        <!-- cheating even more: this means x2t only works when the onlyoffice module is installed
+             and the user is in the onlyoffice group -->
+        <allfonts>/var/lib/onlyoffice/documentserver/server/FileConverter/bin/AllFonts.js</allfonts>
```

Comment:
should we build AllFonts.js as part of the build, or was there a reason that was moved to install time?

Reply:
Possibly the dependency on unfree mscorefonts. Can we maybe get this config file to be read from somewhere other than the package (and especially $out/bin), maybe via an environment variable?

Reply:
Yeah, there should be no config files inside /bin